plugins/module_utils/sops.py

```diff
@@ -59,17 +127,39 @@ class Sops():
     ''' Utility class to perform sops CLI actions '''
 
     @staticmethod
-    def decrypt(encrypted_file, display=None, decode_output=True, rstrip=True, input_type=None, output_type=None):
+    def _add_options(command, env, get_option_value, options):
+        if get_option_value is None:
+            return
+        for option, f in options.items():
+            v = get_option_value(option)
+            if v is not None:
+                f(v, command, env)
+
+    @staticmethod
+    def get_sops_binary(get_option_value):
+        cmd = get_option_value('sops_binary') if get_option_value else None
+        if cmd is None:
+            cmd = 'sops'
+        return cmd
+
+    @staticmethod
+    def decrypt(encrypted_file, display=None, decode_output=True, rstrip=True, input_type=None, output_type=None, get_option_value=None, module=None):
         # Run sops directly, python module is deprecated
-        command = ["sops"]
+        command = [Sops.get_sops_binary(get_option_value)]
+        env = os.environ.copy()
+        Sops._add_options(command, env, get_option_value, GENERAL_OPTIONS)
         if input_type is not None:
             command.extend(["--input-type", input_type])
         if output_type is not None:
             command.extend(["--output-type", output_type])
         command.extend(["--decrypt", encrypted_file])
-        process = Popen(command, stdout=PIPE, stderr=PIPE)
-        (output, err) = process.communicate()
-        exit_code = process.returncode
+
+        if module:
```

Comment:
May you clarify the usage of `module` here? I think I got the idea. Is something made available by `ActionModuleBase`?

Reply:
No, it's for classical `AnsibleModule`s. The Ansible coding guidelines say that modules must never use `subprocess` directly, but use `AnsibleModule.run_command`. This implements that requirement.